### ISSUE TITLE
Java Web Start does not work on Mac OS X

### ISSUE THREAD

**Issue by janvdbergh** [Fedict/eid-client-server#2]
At the moment there is no proper Java Web Start support for Mac OS X:
* JNLP files are not automatically opened, but downloaded instead.
* When opening the file manually it refuses to start because of an unidentified developer warning.

A possible solution would be to sign the JNLP file and bundle it in a zip or dmg archive. See https://stackoverflow.com/questions/19767394/unidentified-developer-for-java-web-start-application-on-os-x for more context.

**Comment by fcorneli** [comment]
Indeed, that's why I never pushed out a WebStart version in the first place...

**Comment by janvdbergh** [comment]
Can you elaborate on alternatives? They might not be implemented here, but can help users of this library to implement their own solution.

**Comment by fcorneli** [comment]
It's now almost 10 years ago that I started working on the eID Applet project. Back then this was a viable solution as it was possible to use a single -what I call- eID runtime across all web browsers and all platforms. Today, this is no longer possible due to:
* high fragmentation of the web browser "common runtime". I.e., the chrome native messaging is not available on all web browsers.
* there is more than desktop these days: Chrome OS, Android, iOS, ...
* now we also have bluetooth smart card readers available

Hence a one-size-fits-all solution is no longer possible unfortunately. One cannot avoid multiple "eID runtimes" to target all web browsers and/or platforms out there.

This impacts how web application vendors can integrate eID within their workflows. Imagine I would open source e-Contract.be completely. Even in that case, it would be unmanageable for our customers to integrate the eID functionality at the eID runtime component level as the maintenance of all these different eID runtimes is way to heavy. Even for us it's a heavy task to keep up with the constantly changing web browser landscape. You simply cannot avoid the "SaaS integration level" anymore. Hence, both public and private sector nowadays cannot avoid a SaaS architecture, where you leave the actual management of all these different eID runtimes to a specialized "entity" like CSAM, e-Contract.be, whatever.

By positioning eid-client-server, I'm afraid the wrong "signal" is given. As if things did not change over the last 10 years...

Even years ago, I already noticed this trend when positioning eID IdP next to the eID Applet. Most developers directly went to eID IdP instead of fighting with the eID Applet. Right now we have like 6 different eID runtimes, so this is certainly the case nowadays.
